src/main/java/core/basesyntax/strategy/DataConverterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Action;
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    public List<FruitTransaction> convertToTransactions(List<String> dateList) {
```

Comment:
The method name `convertToTransactions` should better reflect its purpose. Consider renaming it to something more descriptive, like `parseDataToTransactions` to indicate that it's parsing the input data into transaction objects.